There is a bug in curl 8.6.0 that causes this test to fail.
https://github.com/curl/curl/pull/12981

--- pycurl-7.45.3/tests/option_constants_test.py.orig
+++ pycurl-7.45.3/tests/option_constants_test.py
@@ -458,7 +458,7 @@
 
     # Apparently TLSAUTH_TYPE=SRP is an unknown option on appveyor
     @util.only_unix
-    @util.min_libcurl(7, 52, 0)
+    @util.min_libcurl(8, 7, 0)
     @util.only_ssl_backends('openssl', 'gnutls', 'schannel')
     def test_proxy_tlsauth(self):
         curl = pycurl.Curl()
